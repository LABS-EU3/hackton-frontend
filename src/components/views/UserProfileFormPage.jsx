import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import UserProfileForm from "../templates/UserProfileForm";
import { fetchUserProfile } from "../../store/user/actions";
import UserProfile from "../../components/templates/UserProfile";

const UserProfileFormPage = () => {
  const dispatch = useDispatch();
  const { userId } = useSelector(state => state.currentUser);
  const userDetails = useSelector(state => state.currentUser);
  useEffect(() => {
    dispatch(fetchUserProfile(userId));
  }, [dispatch, userId]);

  console.log("user details", userDetails);
  return (
    <UserProfileForm initialState={userDetails} />
  );
};

export default UserProfileFormPage;
