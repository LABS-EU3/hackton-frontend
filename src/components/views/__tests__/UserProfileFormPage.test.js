import React from 'react'
import { Router, Route } from 'react-router';
import { createMemoryHistory } from 'history';
import { render } from '@testing-library/react'
import UserProfileFormPage from '../userProfileFormPage';
const history = createMemoryHistory()

const renderWithRouter = Component => render(
  <Router history={history}>
    <Route component={Component} />
  </Router>
)

describe("It should render UserProfileFormPage page correctly", () => {
  
    it("renders the UserProfileFormPage component correctly", () => {
      const render = () => renderWithRouter(UserProfileFormPage)
      expect(render).toMatchSnapshot();
    });
  });