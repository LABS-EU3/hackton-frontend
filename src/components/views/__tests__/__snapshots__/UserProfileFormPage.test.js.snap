// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render UserProfileFormPage page correctly renders the UserProfileFormPage component correctly 1`] = `[Function]`;
